cumulusci/core/source_transforms/transforms.py

```diff
@@ -308,8 +315,58 @@ def get_replace_string(self) -> str:
             )
 
 
+class FindReplaceIdAPI(str, enum.Enum):
+    REST = "rest"
+    TOOLING = "tooling"
+
+
+class FindReplaceIdSpec(FindReplaceBaseSpec):
+    replace_record_id_query: str
+    api: FindReplaceIdAPI = FindReplaceIdAPI.REST
+
+    def get_replace_string(self, context: TaskContext) -> str:
+        org = context.org_config
+
+        if self.api is FindReplaceIdAPI.REST:
+            results = org.salesforce_client.query(self.replace_record_id_query)
+        else:
+            results = org.tooling.query(self.replace_record_id_query)
+
+        if results["totalSize"] != 1:
+            raise CumulusCIException(
```

Comment:
@TheBitShepherd @davidmreed  We should standardize the exception messages and error Codes across our product. Probably something we need to look in 244